orttraining/orttraining/core/optimizer/lora_conv1d_replacement.cc

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/optimizer/initializer.h"
+#include "orttraining/core/optimizer/lora_conv1d_replacement.h"
+#include "core/graph/graph_utils.h"
+
+/*
+  in LoRA code, it will use conv1d to do projection for qkv,
+  while the conv1d calculation is mathematically equivalent to matmul, and matmul is much faster than conv1d.
```

Comment:
does this comment apply only to CUDA? , if so, shall we make it clear in the comment? 

"while the conv1d calculation is mathematically equivalent to matmul, and matmul is much faster than conv1d."